migrations/14_refactor_tasks.sql

```diff
@@ -0,0 +1,47 @@
+ALTER TABLE tasks ALTER COLUMN tsk_status TYPE TEXT;
+DROP TYPE task_status;
+
+CREATE TYPE task_status AS ENUM (
+    'success',
+    'pending',
+    'failed'
+);
+
+CREATE TYPE mutability AS ENUM (
+    'immutable',
+    'mutable'
+);
+
+ALTER TABLE tasks RENAME COLUMN tsk_id TO tasks_metadata_hash;
+ALTER TABLE tasks RENAME COLUMN tsk_metadata_url TO tasks_metadata_url;
+ALTER TABLE tasks RENAME COLUMN tsk_error TO tasks_error;
+
+ALTER TABLE tasks
+    ADD COLUMN tasks_task_status task_status NOT NULL DEFAULT 'pending',
+    ADD COLUMN tasks_next_try_at timestamptz DEFAULT NULL,
+    ADD COLUMN tasks_etag text DEFAULT NULL,
+    ADD COLUMN tasks_mutability mutability NOT NULL DEFAULT 'mutable',
+    ADD COLUMN tasks_last_modified_at timestamptz DEFAULT NULL;
+
+UPDATE tasks
+SET tasks_mutability = 'immutable'
+WHERE tasks_metadata_url = ''
+   OR tasks_metadata_url LIKE 'ipfs://%'
+   OR tasks_metadata_url LIKE 'https://ipfs%'
+   OR tasks_metadata_url LIKE 'https://arweave%'
+   OR tasks_metadata_url LIKE 'https://www.arweave%';
+
+UPDATE tasks SET tasks_next_try_at = NOW() WHERE tasks_task_status = 'success';
```

Comment:
Given it's a new column 
    ADD COLUMN tasks_task_status task_status NOT NULL DEFAULT 'pending',
it'll not have success state ever. We'll also loose the status, is that expected?